test/libsolidity/semanticTests/functionCall/return_size_expectation_mismatch.sol

```diff
@@ -0,0 +1,31 @@
+interface ShortReturn {
+    function f() external pure returns (bytes32);
+}
+contract LongReturn {
+    function f() external pure returns (uint[20] memory) {}
+}
+
+contract Test {
+    function test() public returns (uint) {
+        LongReturn lr = new LongReturn();
+        uint freeMemoryBefore;
+        assembly {
+            freeMemoryBefore := mload(0x40)
+        }
+
+        lr.f();
```

Comment:
You have to convert it to a `ShortReturn` before you call it so that the compiler thinks it is returning just a single `bytes32`